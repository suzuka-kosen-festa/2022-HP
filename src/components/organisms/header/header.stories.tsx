import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { ComponentPropsWithoutRef } from "react";
import tw from "twin.macro";
import { Header } from ".";

type T = typeof Header;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  headerLinks: [
    {
      link: "/",
      color: tw`text-primary-yellow`,
      name: "Top",
    },
    {
      link: "/member",
      color: tw`text-primary-red`,
      name: "Members",
    },
    {
      link: "/timetable",
      color: tw`text-primary-green`,
      name: "Schedule",
    },
    {
      link: "/bazaar",
      color: tw`text-primary-purple`,
      name: "Bazaar",
    },
    {
      link: "/sponsors",
      color: tw`text-primary-blue`,
      name: "Sponsors",
    },
  ],
};

export default {
  args,
  argTypes: {},
  component: Header,
} as Meta;

export const Default: Story = {};
