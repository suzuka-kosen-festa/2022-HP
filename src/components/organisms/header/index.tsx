import type { FC } from "react";
import tw, { css } from "twin.macro";
import { Link } from "wouter";
import type { HeaderProperties, HeaderLinkItemProperties } from "./types/model";
import { AnchorText } from "../../atoms/anchor-text";
import { HamburgerMenu } from "../../molecules/hamburger-menu";

const HeaderContainer = tw.header`fixed flex bg-center bg-cover justify-center px-header-x py-header-y md:(justify-between px-header-x-md py-header-y-md)`;

const Logo = tw.img`w-min cursor-pointer`;

const LinkWrapper = tw.ul`hidden md:flex space-x-10 justify-center items-center`;

const LinkItem: FC<HeaderLinkItemProperties> = ({ children, link, color }) => (
  <li role="menuitem">
    <Link to={link}>
      <AnchorText css={[tw`md:text-base lg:text-2xl  hover:underline`, color]}>{children}</AnchorText>
    </Link>
  </li>
);

const Header: FC<HeaderProperties> = ({ headerLinks }) => (
  <HeaderContainer
    css={css`
      background-image: url("/statics/header-md.webp");
      aspect-ratio: 48/ 7;
      @media (min-width: 768px) {
        background-image: url("/statics/header.webp");
        aspect-ratio: 12 / 1;
      }
    `}
  >
    <Link to="/">
      <Logo
        src="/statics/logo.png"
        alt="第57回鈴鹿高専祭、Re:ROADのロゴ"
        width={948}
        height={197}
        loading="eager"
      />
    </Link>
    <LinkWrapper>
      {headerLinks.map(headerLink => (
        <LinkItem link={headerLink.link} color={headerLink.color} key={headerLink.name}>
          {headerLink.name}
        </LinkItem>
      ))}
    </LinkWrapper>
    <HamburgerMenu menuLinks={headerLinks} />
  </HeaderContainer>
);

export { Header };
