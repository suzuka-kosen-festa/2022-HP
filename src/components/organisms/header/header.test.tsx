import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./header.stories";

const { Default } = composeStories(stories);

const banneroptions = {
  name: "",
};

const imgoptions = {
  name: "第57回鈴鹿高専祭、Re:ROADのロゴ",
};

describe("(components) organisms/header", () => {
  test("to be organisms", () => {
    const { container } = render(<Default />);
    expect(container).toBeOrganism();
  });
  test("to be [role=banner]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("banner", banneroptions)).toBeInTheDocument();
  });
  test("to be [role=img]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("img", imgoptions)).toBeInTheDocument();
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
