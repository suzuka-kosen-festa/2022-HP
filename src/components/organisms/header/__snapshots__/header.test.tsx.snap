// Vitest Snapshot v1

exports[`(components) organisms/header > take snap shot 1`] = `
<div>
  <header
    class="css-ojgxcq-Header"
  >
    <img
      alt="第57回鈴鹿高専祭、Re:ROADのロゴ"
      class="css-pfnskm"
      height="197"
      href="/"
      loading="eager"
      src="/statics/logo.png"
      width="948"
    />
    <ul
      class="css-1ptipdp"
    >
      <li
        role="menuitem"
      >
        <a
          class="css-yzajyk-LinkItem"
          href="/"
        >
          Top
        </a>
      </li>
      <li
        role="menuitem"
      >
        <a
          class="css-s8xoeo-LinkItem"
          href="/member"
        >
          Members
        </a>
      </li>
      <li
        role="menuitem"
      >
        <a
          class="css-dgtzi5-LinkItem"
          href="/timetable"
        >
          Schedule
        </a>
      </li>
      <li
        role="menuitem"
      >
        <a
          class="css-1lc6xq8-LinkItem"
          href="/bazaar"
        >
          Bazaar
        </a>
      </li>
      <li
        role="menuitem"
      >
        <a
          class="css-zzwddn-LinkItem"
          href="/sponsors"
        >
          Sponsors
        </a>
      </li>
    </ul>
    <div
      class="css-1wlf7oa"
    >
      <button
        aria-controls="hamburger-list"
        aria-expanded="false"
        aria-haspopup="menu"
        aria-label="menu button"
        class="css-cj7h3e"
      >
        <div
          class="css-17dtj6s"
        >
          <span />
        </div>
      </button>
      <div />
    </div>
  </header>
</div>
`;
